Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -46,23 +46,22 @@ @implementation SentryMetricReading
  */
 SentrySerializedMetricEntry *_Nullable serializeValuesWithNormalizedTime(
     NSArray<SentryMetricReading *> *absoluteTimestampValues, NSString *unit,
-    SentryTransaction *transaction)
```

Comment:
Same thing here with type safety, in addition to cutting down the amount of parameters. Unless this is required for this PR due to some nuance with compilation for hybrid platforms, I'd prefer this be reverted.